packages/nc-gui-v2/components/smartsheet/sidebar/toolbar/index.vue

```diff
@@ -1,14 +1,76 @@
 <script lang="ts" setup>
+import FileSaver from 'file-saver'
+import { message } from 'ant-design-vue'
 import AddRow from './AddRow.vue'
 import LockMenu from './LockMenu.vue'
 import Reload from './Reload.vue'
 
 const { isUIAllowed } = useUIPermission()
+
+const { api } = useApi()
+
+const debug = $ref(false)
+
+const clickCount = $ref(0)
+
+async function exportCache() {
+  try {
+    const data = await api.utils.cacheGet()
+    if (!data) {
+      message.info('Cache is empty')
+      return
+    }
+    const blob = new Blob([JSON.stringify(data)], {
+      type: 'text/plain;charset=utf-8',
+    })
+    FileSaver.saveAs(blob, 'cache_exported.json')
+    message.info('Exported Cache Successfully')
+  } catch (e: any) {
+    message.error(e.message)
+  }
+}
+
```

Comment:
All the changes here could be moved into a separate component so we keep the toolbar wrapper clean.

Reply:
Agree.